app/repositories/application/ReportingRepoBSONToModelHelper.scala

```diff
@@ -0,0 +1,250 @@
+    val disability = adDoc.flatMap(_.getAs[String]("hasDisability"))
+    val onlineAdjustments = adDoc.flatMap(_.getAs[Boolean]("needsSupportForOnlineAssessment")).map(booleanTranslator)
+    val assessmentCentreAdjustments = adDoc.flatMap(_.getAs[Boolean]("needsSupportAtVenue")).map(booleanTranslator)
+    val gis = adDoc.flatMap(_.getAs[Boolean]("guaranteedInterview"))
+
+    val civilServiceExperienceDoc = doc.getAs[BSONDocument]("civil-service-experience-details")
+    val civilServiceExperience = toCivilServiceExperienceDetailsReportItem(civilServiceExperienceDoc)
+
+    val applicationId = doc.getAs[String]("applicationId").getOrElse("")
+    val userId = doc.getAs[String]("userId").getOrElse("")
+    val progress: ProgressResponse = toProgressResponse(doc, applicationId)
+
+    ApplicationForDiversityReport(applicationId, userId, Some(ProgressStatusesReportLabels.progressStatusNameInReports(progress)),
+      schemes.getOrElse(List.empty), disability, gis, onlineAdjustments,
+      assessmentCentreAdjustments, civilServiceExperience)
+  }
+
+  def toApplicationForOnlineTestPassMarkReport(doc: BSONDocument): ApplicationForOnlineTestPassMarkReport = {
+
+    val applicationId = doc.getAs[String]("applicationId").getOrElse("")
+
+    val schemesDoc = doc.getAs[BSONDocument]("scheme-preferences")
+    val schemes = schemesDoc.flatMap(_.getAs[List[SchemeType]]("schemes"))
+
+    val adDoc = doc.getAs[BSONDocument]("assistance-details")
+    val gis = adDoc.flatMap(_.getAs[Boolean]("guaranteedInterview"))
+    val disability = adDoc.flatMap(_.getAs[String]("hasDisability"))
+    val onlineAdjustments = adDoc.flatMap(_.getAs[Boolean]("needsSupportForOnlineAssessment")).map(booleanTranslator)
+    val assessmentCentreAdjustments = adDoc.flatMap(_.getAs[Boolean]("needsSupportAtVenue")).map(booleanTranslator)
+
+    val progress: ProgressResponse = toProgressResponse(doc, applicationId)
+
+    ApplicationForOnlineTestPassMarkReport(
+      applicationId,
+      ProgressStatusesReportLabels.progressStatusNameInReports(progress),
+      schemes.getOrElse(Nil),
+      disability,
+      gis,
+      onlineAdjustments,
+      assessmentCentreAdjustments,
+      toTestResultsForOnlineTestPassMarkReportItem(doc, applicationId))
+  }
+
+  private[application] def toTestResultsForOnlineTestPassMarkReportItem(doc: BSONDocument, applicationId: String) = {
+
+    val testGroupsDoc = doc.getAs[BSONDocument]("testGroups")
+    val (behaviouralTestResult, situationalTestResult) = toPhase1TestResults(testGroupsDoc)
+    val etrayTestResult = toPhase2TestResults(applicationId, testGroupsDoc)
+    val videoInterviewResults = toPhase3TestResults(testGroupsDoc)
+
+      TestResultsForOnlineTestPassMarkReportItem(behaviouralTestResult, situationalTestResult, etrayTestResult, videoInterviewResults)
+  }
+
+  private[application] def toPhase1TestResults(testGroupsDoc: Option[BSONDocument]) = {
+    val phase1Doc = testGroupsDoc.flatMap(_.getAs[BSONDocument]("PHASE1"))
```

Comment:
Which enum? Bear in mind this "PHASE1" is in the context of JSON, it is not necessarily tied up with the status.